modules/mobile/app/models/mobile/v2/adapters/appointments.rb

```diff
@@ -0,0 +1,328 @@
+          return nil if requested_periods.nil?
+
+          proposed_times = []
+          requested_periods.each do |period|
+            start_date = DateTime.parse(period[:start])
+            proposed_times << {
+              date: start_date.strftime('%m/%d/%Y'),
+              time: start_date.hour.zero? ? 'AM' : 'PM'
+            }
+          end
+
+          proposed_times
+        end
+
+        def vetext_id(appointment_hash, start_date_local)
+          "#{appointment_hash[:clinic]};#{start_date_local.strftime('%Y%m%d.%H%S%M')}"
+        end
+
+        def status(appointment_hash)
+          return STATUSES[:hidden] if HIDDEN_STATUS.include?(appointment_hash[:status])
+
+          STATUSES[appointment_hash[:status].to_sym]
+        end
+
+        def start_date_utc(appointment_hash)
+          start = appointment_hash[:start] || appointment_hash.dig(:requested_periods, 0, :start)
+          return nil if start.nil?
+
+          DateTime.parse(start)
+        end
+
+        def parse_by_appointment_type(appointment_hash, type)
+          case type
+          when 'phone', 'clinic'
+            APPOINTMENT_TYPES[:va]
+          when 'cc'
+            APPOINTMENT_TYPES[:cc]
+          when 'telehealth'
+            if appointment_hash.dig(:telehealth, :vvs_kind) == VIDEO_GFE_CODE
+              APPOINTMENT_TYPES[:va_video_connect_gfe]
+            elsif appointment_hash.dig(:telehealth, :atlas)
+              APPOINTMENT_TYPES[:va_video_connect_atlas]
+            else
+              APPOINTMENT_TYPES[:va_video_connect_home]
+            end
+          end
+        end
+
+        # rubocop:disable Metrics/MethodLength
+        def location(type, appointment_hash)
+          location = {
+            id: nil,
+            name: nil,
+            address: {
+              street: nil,
+              city: nil,
+              state: nil,
+              zip_code: nil
+            },
+            lat: nil,
+            long: nil,
+            phone: {
+              area_code: nil,
+              number: nil,
+              extension: nil
+            },
+            url: nil,
+            code: nil
+          }
+          telehealth = appointment_hash[:telehealth]
+
+          case type
+          when APPOINTMENT_TYPES[:cc]
+            cc_location = appointment_hash.dig(:extension, :cc_location)
+
+            if cc_location.present?
+              location[:name] = cc_location[:practice_name]
+              location[:address] = {
+                street: cc_location.dig(:address, :line)&.join(' ')&.strip,
+                city: cc_location.dig(:address, :city),
+                state: cc_location.dig(:address, :state),
+                zip_code: cc_location.dig(:address, :postal_code)
+              }
+            end
+          when APPOINTMENT_TYPES[:va_video_connect_atlas],
+            APPOINTMENT_TYPES[:va_video_connect_home],
+            APPOINTMENT_TYPES[:va_video_connect_gfe]
+            if telehealth
```

Comment:
that's one of the big question marks I have. no documentation answers those questions and while I've been able to find a couple appointments that exist with the same data (or nearly the same to be confident that they're the same) on both v0 and v2, I haven't found that with telehealth appointments, so I tried to make it so if the field is there, then it'll use the data. I think it's safe to assume that if the v2 appointment_kind is `telehealth` then it'll be one of these. It's not pretty but we may just want to address this in the follow up ticket when we get more context on telehealth.

Reply:
Fair enough. We'll probably have to figure this our in staging with the help of QA. Now that you have a review instance, shouldn't it be possible to flip the flag for this user and compare v0 and v2 results? There may be some issues with some of the CC requests being missing because they're on the wrong server, but it still might help us figure out some issues.